wayland-server/src/lib.rs

```diff
@@ -1,4 +1,72 @@
+//! provides the types generated from the core protocol in the [`protocol`] module. For other standard
+//! protocols, see the `wayland-protocols` crate.
+//!
+//! ## Request dispatching and the [`Dispatch`] trait
+//!
+//! The request dispatching logic provided by this crate is build around the [`Dispatch`] trait. During the
+//! dispatching process (in [`Display::dispatch_clients()`]), all requests sent by clients are read from
+//! their respective process and delivered to your processing logic, by invoking methods on the various
+//! [`Dispatch`] implementations of your `State` struct. In this paradigm, your `State` needs to implement
+//! `Dispatch<O, _>` for every Wayland object `O` it needs to process events for.
+//!
+//! However, implementing all those traits on your own is a lot of (often uninteresting) work. To make this
+//! easier a composition mechanism is provided using the [`delegate_dispatch!`] macro. This way, another
+//! library (such as Smithay) can provide generic [`Dispatch`] immplementations that you can reuse on your
+//! own app by delegating thos objects to that provided implementation. See the documentation of those traits
+//! and macro for details.
+//!
+//! ## Globals
+//!
+//! The entry point of the protocol for clients goes through the protocol globals. Each global represents a
+//! capability of your compositor, a peripheral it has access to, or a protocol extension it supports. Those
+//! globals are created by you using [`DisplayHandle::create_global()`], and require yourn `State` to
+//! implement the [`GlobalDispatch`] trait for the interface associated with that global.
+//!
+//! ## Advanced use
+//!
+//! ### Bypassing [`Dispatch`]
+//!
+//! It may be that for some of your objects, handling them via the [`Dispatch`] trait is unpractical. In
+//! those contexts, this crate also provides some escape-hatches to directly interface with the low-level
+//! APIs from `wayland-backend`, allowing you to register callbacks for those objects by directly providing
+//! implementations of the backend [`ObjectData`](crate::backend::ObjectData) trait.
+//! See [`Client::create_resource_from_objdata()`] and [`DataInit::custom_init()`].
+//!
+//! ### Interaction with FFI
+//!
+//! It can happen that you'll need to interact with Wayland states accross FFI, such as for example when
+//! interfacing with the graphics stack for enabling hardware acceleration for clients.
+//!
+//! In this case, you'll need to do it in two steps, by explicitly working with `wayland-backend`, adding
+//! it to your dependencies and enabling its `server_system` feature.
+//!
+//! Then, you'll generally need:
+//!
+//! - The `*mut wl_display` pointer, that you can retrieve by first retrieving the
+//!   [`Backend`](crate::backend::Backend) using [`Display::backend()`], and then invoke
+//!   [`Backend::display_ptr()](crate::backend::Backend::display_ptr).
```

Comment:
Are the docs built with the sys backend? (Regarding the warning)

Reply:
I'll check how to ensure that for CI. On docs.rs it should be good, `wayland-backend` is documented with all its features enabled.

Reply:
Ugh, looks like cargo does not provide a simple way to force that actually. I'll remove the link then, it'll never work anyway.